rclpy/rclpy/node.py

```diff
@@ -761,6 +763,25 @@ def _set_parameters_atomically(
 
         return result
 
+    def add_on_set_paramters_callback(
+        self,
+        callback: Callable[[List[Parameter]], SetParametersResult]
+    ) -> None:
+        """Add the callback to list."""
+        prevCallBack = self._parameters_callback
+        prevCallBack.insert(0, callback)
+        self._parameters_callback = prevCallBack
+
+    def remove_on_set_parameters_callback(
+        self,
+        callback: Callable[[List[Parameter]], SetParametersResult]
+    ) -> None:
+        """Remove callback from list."""
+        if callback in self._parameters_callback:
+            self._parameters_callback.remove(callback)
+        else:
+            print('Callback does not exist')
```

Comment:
Error messages should in general be printed to `stderr` (not `stdout`). In this case I don't think printing an error message is good - instead the function should raise an exception (and document that in the docblock).

Reply:
@dirk-thomas sir I dont know how to add the testcases,I tried understanding the test_node.py but cant seem to.